test/automation/src/terminal.ts

```diff
@@ -9,38 +9,62 @@ import { QuickAccess } from './quickaccess';
 
 const TERMINAL_VIEW_SELECTOR = `#terminal`;
 const XTERM_SELECTOR = `${TERMINAL_VIEW_SELECTOR} .terminal-wrapper`;
-const CONTRIBUTED_PROFILE_NAME = `JavaScript Debug Terminal`;
 const TABS = '.tabs-list .terminal-tabs-entry';
 const XTERM_FOCUSED_SELECTOR = '.terminal.xterm.focus';
+const PLUS_BUTTON_SELECTOR = '.codicon-plus';
+const EDITOR_GROUPS_SELECTOR = '.editor .split-view-view';
+const EDITOR_TAB_SELECTOR = '.terminal-tab';
+const SINGLE_TAB_SELECTOR = '.single-terminal-tab';
 
-const enum TerminalCommandId {
+export enum TerminalCommandIdWithValue {
 	Rename = 'workbench.action.terminal.rename',
 	ChangeColor = 'workbench.action.terminal.changeColor',
 	ChangeIcon = 'workbench.action.terminal.changeIcon',
+	NewWithProfile = 'workbench.action.terminal.newWithProfile',
+	SelectDefaultProfile = 'workbench.action.terminal.selectDefaultShell'
+}
+
+export enum TerminalCommandId {
 	Split = 'workbench.action.terminal.split',
 	KillAll = 'workbench.action.terminal.killAll',
 	Unsplit = 'workbench.action.terminal.unsplit',
 	Join = 'workbench.action.terminal.join',
 	Show = 'workbench.action.terminal.toggleTerminal',
 	CreateNew = 'workbench.action.terminal.new',
+	CreateNewEditor = 'workbench.action.createTerminalEditor',
+	SplitEditor = 'workbench.action.createTerminalEditorSide',
+	MoveToPanel = 'workbench.action.terminal.moveToTerminalPanel',
+	MoveToEditor = 'workbench.action.terminal.moveToEditor',
 	NewWithProfile = 'workbench.action.terminal.newWithProfile',
 	SelectDefaultProfile = 'workbench.action.terminal.selectDefaultShell'
 }
+interface TerminalLabel {
+	name?: string,
```

Comment:
We should add some jsdoc here to explain `'*'` is supported as a wildcard when used in assertions.

Also while it doesn't do anything with the compile, I think using `: string | '*'` will have it show up in intellisense so it's more obvious to users of the interface.

Reply:
I'll remove the ANY_NAME code for now bc contributed profile tests are skipped and the names are covered by https://github.com/microsoft/vscode/blob/bc75fdc7e3220ed62799be00908514399a1e2fcd/test/automation/src/terminal.ts#L84
already

Reply:
making a note on here https://github.com/microsoft/vscode/issues/137392